traits/tests/test_editor_factories.py

```diff
@@ -207,3 +207,11 @@ def test_list_editor_list_instance(self):
         trait = List(Instance(HasTraits))
         editor = list_editor(trait, trait)
         self.assertIsInstance(editor, traitsui.api.TableEditor)
+
+    # regression test for enthought/traitsui#1539
+    def test_list_editor_list_instance_row_factory(self):
+        trait = List(Instance(HasTraits, kw={}))
+        editor = trait.create_editor()
```

Comment:
I changed this line as I thought `list_editor(trait, trait)` was more confusing to read, but I now realize it was simply copied from the other tests.  I am happy to revert if that is preferred